internal/restorable/image.go

```diff
@@ -722,15 +724,23 @@ func regionToRectangle(region graphicsdriver.Region) image.Rectangle {
 
 // removeDuplicatedRegions removes duplicated regions and returns the new size of the slice.
 // If a region covers other regions, the covered regions are removed.
-func removeDuplicatedRegions(regions []image.Rectangle) int {
+// This assumes for efficiency that no duplicated regions exist in regions[0:firstAdded].
+func removeDuplicatedRegions(regions []image.Rectangle, firstAdded int) int {
```

Comment:
I feel like the name `firstAdded` seems confusing. Is there a better name?

Reply:
What about `alreadyCheckedUntil`? (Disclaimer: I am not an English native speaker)

Reply:
Renamed.